extensions/notebook/src/jupyter/jupyterSessionManager.ts

```diff
@@ -308,7 +308,7 @@ export class JupyterSession implements nb.ISession {
 				}
 			}
 			else {
-				connectionProfile.options[KNOX_ENDPOINT_PORT] = this.getKnoxPortOrDefault(connectionProfile);
+				return Promise.reject(new Error(localize('providerNotValid', "Non-MSSQL providers are not supported for spark kernels.")));
```

Comment:
This function is async so you can just throw - no need to wrap it in a Promise.reject